pages/_app.js

```diff
@@ -94,15 +75,26 @@ export default function App({ Component, pageProps }) {
     }
   }
 
-  // Search Dictionary
+  // Search Dictionary and check if word is already in Word List
   useEffect(() => {
     setDictionaryResults([]);
 
     if (data) {
       const structuredOutput = refactorDictionaryOutput(data);
-      setDictionaryResults(structuredOutput);
+      const checkedOutput = structuredOutput.map((entry) => {
+        const isInWordList = wordList.find(
+          (wordListEntry) => wordListEntry.slug === entry.slug
+        );
+        if (isInWordList) {
+          return { ...entry, showAddButton: false };
+        }
+        return entry;
+      });
+
+      // setDictionaryResults(structuredOutput);
+      setDictionaryResults(checkedOutput);
     }
-  }, [dictionaryQuery, data]);
+  }, [dictionaryQuery, data, wordList]);
```

Comment:
I don't think you need a useEffect for this. You could a handlerFunction (such as handleDictionaryResults) that contains all of the logic that is currently part of your useEffect.

Instead of passing the setDictionaryResults function down as a prop, you pass down your newly created handleDictionaryResults.

```suggestion
function handleDictionaryResults() {
    setDictionaryResults([]);

    if (data) {
      const structuredOutput = refactorDictionaryOutput(data);
      const checkedOutput = structuredOutput.map((entry) => {
        const isInWordList = wordList.find(
          (wordListEntry) => wordListEntry.slug === entry.slug
        );
        if (isInWordList) {
          return { ...entry, showAddButton: false };
        }
        return entry;
      });

      // setDictionaryResults(structuredOutput);
      setDictionaryResults(checkedOutput);
    }
  }
```

Reply:
I used a function for the check in the useEffect. I do think I need the useEffect for the dictionary results to work, because the "data" is coming from the api via swr - I tried without it and it didn't refresh the dictionary data without refreshing. But I actually already had the function that checks the word list in the utils I just didn't use it 😅 I added that now.

Reply:
ah, I see, then I must have missed some parts of the flow of your app :-)